### PR TITLE
Adding Step Status [minor bump]

### PR DESCRIPTION
**Pull request by stjohnjohnson**
This update does the following:
 - writes the status to the API 
 - switch to using gitversion
 - fix checkout location to include hostname
 - allow override of emitter location

Feature: https://github.com/screwdriver-cd/screwdriver/issues/135

### CODE REVIEW

**Comment by stjohnjohnson** [comment]
PS. the Screwdriver build will continue to fail until we fix the GoPath issue.

File: executor/executor.go [hunk 1]
```diff
@@ -20,36 +20,54 @@ func (e ErrStatus) Error() string {
...
+	fmt.Fprintf(emitter, "$ %s\n", cmd.Cmd)
+	emitter.StartCmd(cmd)
+	c.Stdout = emitter
+	c.Stderr = emitter
+
+	c.Dir = path
+	c.Env = append(env, c.Env...)
+
+	if err := c.Start(); err != nil {
+		return 255, fmt.Errorf("launching command %q: %v", cmd.Cmd, err)
```
Comment (jer) [inline executor/executor.go]: These numbers would be nice as constants

File: launch_test.go [hunk 2]
```diff
@@ -48,18 +59,18 @@ func (r MockRepo) GetPath() string {
 	return ""
 }
 
-func mockAPI(t *testing.T, testBuildID, testJobID, testPipelineID string, testStatus screwdriver.BuildStatus) MockAPI {
+func mockAPI(t *testing.T, TestBuildID, TestJobID, testPipelineID string, testStatus screwdriver.BuildStatus) MockAPI {
```
Comment (jer) [inline launch_test.go]: Did you do this on purpose? Masking the consts above?

File: screwdriver/screwdriver_test.go [hunk 3]
```diff
@@ -376,5 +377,40 @@ func TestUpdateBuildStatus(t *testing.T) {
...
+		want := regexp.MustCompile(`{"startTime":"[\d-]+T[\d:.Z-]+"}`)
+		if !want.MatchString(buf.String()) {
+			t.Errorf("buf.String() = %q", buf.String())
+		}
+	})
+	testAPI := api{"http://fakeurl", "faketoken", http}
 
+	err := testAPI.UpdateStepStart("step1", "build1")
+
+	if !reflect.DeepEqual(err, nil) {
```
Comment (jer) [inline screwdriver/screwdriver_test.go]: Huh?

File: executor/executor.go (hunk 1 above)
Comment (stjohnjohnson) [inline executor/executor.go]: Good point, will do.

File: launch_test.go (hunk 2 above)
Comment (stjohnjohnson) [inline launch_test.go]: Mistake, will correct

**Comment by stjohnjohnson** [comment]
@jer updated

**Comment by jer** [comment]
👍  beautiful

**Pull request merged**